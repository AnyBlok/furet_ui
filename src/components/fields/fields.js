export const fields = {
  list: {},
  form: {},
  thumbnail: {},
  kanban: {},
};
export default {
    fields,
}
