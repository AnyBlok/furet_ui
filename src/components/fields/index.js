import { fields } from './fields';
import './common';
import './boolean';
import './datetime';
import './integer';
import './password';
import './timestamp';
import './selection';
import './string';

import './relationship';

import { defineComponent } from '../factory';

defineComponent('furet-ui-unknown-field', {
  template : `<div>Unknow Field <strong>{{config.type}}</strong></div>`,
  prototype: {
    props: ['config', 'data'],
  }
})

defineComponent('furet-ui-field', {
  template : `
    <component 
      v-bind:is="component" 
      v-bind:resource="resource" 
      v-bind:config="config" 
      v-bind:data="data"
    />
  `,
  prototype: {
    props: ['resource', 'config', 'data'],
    computed: {
      component () {
        return (fields[this.resource.type] || {})[this.config.type] || 'furet-ui-unknown-field'
      },
    },
  }
})
