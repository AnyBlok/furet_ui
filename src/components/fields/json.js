/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import {defineComponent} from '../factory'
import {fields} from './fields';

const format = (value) => {
  if (value)
    try {
        let parsed = value;
        if (typeof value === "string") {
            parsed = JSON.parse(value)
        }
      return JSON.stringify(parsed, null, 2);
    } catch (e) {
      // console.error('Fields.Json : parse json', e)
      return null
    }
  return value;
}


/**
 * furet-ui-list-field-json component is used to manage json/jsonb on list
 * resource (``furet-ui-resource-list``).
 *
 * @example
 *  <furet-ui-form-field-json
 *    :config="aConfigObject"
 *    :data="aDataObject"
 *    :resource="aResourceObject"/>
 *
 * @param {Object} config - A config object to manage the behaviour of the component
 * @param {Object} data - An object that contains data to display. The key to use
 *                        in set in the `config.key`
 * @param {Object} resource - A resource object used to properly bind data with parents
 *                            tags and manage reactivity.
 *
 * ``config`` Object contains
 * @param {String} name - the key to use in the ``data`` object where is store the value
 */
defineComponent('furet-ui-list-field-json', {
  template: `
    <div>
      <span v-if="isHidden" />
      <pre 
        v-else
        v-bind:style="{width: '100%', padding: 2, backgroundColor: 'white'}"
      >{{value}}</pre>
    </div>`,
  extend: ['furet-ui-list-field-common'],
  prototype: {
    computed: {
      value () {
        const value = this.data[this.config.name] || '';
        return format(value)
      },
    },
  },
})
fields.list.json = 'furet-ui-list-field-json'
fields.list.jsonb = 'furet-ui-list-field-json'

defineComponent('furet-ui-thumbnail-field-json', {
  template: `
    <furet-ui-thumbnail-field-common-tooltip-field
      v-bind:resource="resource"
      v-bind:data="data"
      v-bind:config="config"
    >
      <pre 
        v-bind:style="{width: '100%', padding: 2, backgroundColor: 'white'}"
      >{{value}}</pre>
    </furet-ui-thumbnail-field-common-tooltip-field>
  `,
  extend: ['furet-ui-thumbnail-field-common'],
  prototype: {
    computed: {
      value () {
        const value = this.data[this.config.name] || '';
        return format(value)
      },
    },
  },
})
fields.thumbnail.json = 'furet-ui-thumbnail-field-json'
fields.thumbnail.jsonb = 'furet-ui-thumbnail-field-json'
fields.kanban.json = 'furet-ui-thumbnail-field-json'
fields.kanban.jsonb = 'furet-ui-thumbnail-field-json'

/**
 * furet-ui-form-field-json component is used to manage json/jsonb on form
 * resource (``furet-ui-resource-form``).
 *
 * @example
 *  <furet-ui-form-field-json
 *    :config="aConfigObject"
 *    :data="aDataObject"
 *    :resource="aResourceObject"/>
 *
 * @param {Object} config - A config object to manage the behaviour of the component
 * @param {Object} data - An object that contains data to display. The key to use
 *                        in set in the `config.key`
 * @param {Object} resource - A resource object used to properly bind data with parents
 *                            tags and manage reactivity.
 *
 * ``config`` Object contains
 * @param {String} name - the key to use in the ``data`` object where is store the value
 * @param {String?} placeholder - A placeholder to help user to know what to collect
 */
defineComponent('furet-ui-form-field-json', {
  template: `
    <furet-ui-form-field-common-tooltip
      v-bind:resource="resource"
      v-bind:data="data"
      v-bind:config="config"
    >
      <b-field 
        v-bind:label="config.label"
        v-bind:type="getType"
        v-bind:message="getMessage"
        v-bind:style="{'width': 'inherit'}"
      >
        <pre v-if="isReadonly"
            v-bind:style="{width: '100%', padding: 2, backgroundColor: 'white'}"
        >{{value}}</pre>
        <b-input 
          type="textarea"
          v-else 
          v-bind:value="value" 
          v-on:input="updateValue"
          v-bind:placeholder="config.placeholder"
        />
      </b-field>
    </furet-ui-form-field-common-tooltip>
  `,
  extend: ['furet-ui-form-field-common', 'furet-ui-form-field-common-tooltip-field'],
  prototype: {
    computed: {
      raw_value () {
        return this.data[this.config.name] || '';
      },
      value () {
        return format(this.raw_value)
      },
      getType () {
        if (this.isRequired) {
          if (this.raw_value) {
            if (this.isJsonInvalid) return 'is-danger';
            return 'is-info';
          }
          return 'is-danger';
        }
        if (this.isJsonInvalid) return 'is-danger';
        return '';
      },
      getMessage () {
        if (this.isRequired) {
          if (!this.value) {
            if (this.isJsonInvalid) return this.$t('components.fields.json.invalid');
            return this.$i18n.t('components.fields.common.required');
          }
        }
        if (this.isJsonInvalid) return this.$t('components.fields.json.invalid');
        return ''
      },
      isJsonInvalid () {
        if (!this.isReadonly) {
          if (! this.raw_value) return false
          try {
            JSON.parse(this.raw_value)
          } catch (e) {
            // console.error(e, this.raw_value)
            return true;
          }
        }
        return false;
      },
    },
  },
})
fields.form.json = 'furet-ui-form-field-json'
fields.form.jsonb = 'furet-ui-form-field-json'
