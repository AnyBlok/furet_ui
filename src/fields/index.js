/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import React from 'react';
import plugin from '../plugin';
import './string';
import './selection';
import './date';
import './integer';
import './float';
import './url';
import './uuid';
import './sequence';
import './password';
import './color';
import './text';

/**
 * Unknown field, used only if the wanted field is not available
**/
plugin.set(['field'], {Unknown: (props) => {
    return <div>{props.value}</div>
}});

/**
 * return the component for viewType and fieldType
**/
export const getField = (viewType, fieldType, attribs, value) => {
    let field = plugin.get(['field', viewType, fieldType]);
    if (!field) {
        field = plugin.get(['field', 'Unknown']);
        console.log('getField', viewType, fieldType, attribs, value)
    }
    const id = 'view-type-' + viewType + '-field_type-' + fieldType + '-field-name-' + (attribs.name || '');
    return React.createElement(field, Object.assign({id}, attribs, {value}));
}

export default {
    getField,
}
