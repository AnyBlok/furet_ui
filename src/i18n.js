/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueI18n from 'vue-i18n';
Vue.use(VueI18n);


export const i18nConf = {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
            fields: {
                common: {
                    required: 'this field is required',
                },
                json: {
                    invalid: 'Invalid JSON format',
                },
                file: {
                    upload: 'Upload',
                    download: 'Download',
                    delete: 'Delete',
                },
            },
        },
    },
};
export const i18n = new VueI18n(i18nConf);
