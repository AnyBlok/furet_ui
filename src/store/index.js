/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import _ from 'underscore';
import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import * as getters from './getters';
import modules from './modules';
import {i18n} from '../i18n';
import {router} from '../routes';
import moment from 'moment';

Vue.use(Vuex)
const debug = process.env.NODE_ENV !== 'production'

export const storeDef = {
    actions,
    getters,
    modules,
    strict: debug,
};

export const store = new Vuex.Store(storeDef);

export const dispatchAll = (datas) => {
    _.each(datas, data => {
        if (data.type) {
            switch (data.type) {
                case 'UPDATE_LOCALES':
                    _.each(data.locales, locale => {
                        i18n.setLocaleMessage(locale.locale, locale.messages)
                    });
                    break
                case 'SET_LOCALE':
                    i18n.locale = data.locale;
                    moment.locale(data.locale);
                    break
                case 'UPDATE_ROUTE':
                    if (data.name) router.push({name: data.name, params: data.params});
                    else if (data.path) router.push({path: data.path});
                    break;
                default:
                    const type = data.type;
                    delete data.type;
                    const process = data.process || 'commit';
                    store[process](type, data);
            }
        }
    });
}

export default store;
