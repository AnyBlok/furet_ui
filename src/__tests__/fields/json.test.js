/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import Vuex from 'vuex';
import Buefy from 'buefy';
Vue.use(Buefy, {defaultIconPack: 'fa',});
import {store} from '../../store';
import {router} from '../../routes';
import {i18n} from '../../i18n';
import {FieldListJson, FieldThumbnailJson, FieldFormJson} from '../../fields/json'

describe('Json list component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render with data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListJson, {props: {
                row: {fieldname: '{"a": "b"}'},
                header: {name: 'fieldname'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListJson, {props: {
                row: {},
                header: {name: 'fieldname'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListJson, {props: {
                row: {fieldname: '{"a": "b"}'},
                header: {name: 'fieldname', invisible: true},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListJson, {props: {
                row: {fieldname: '{"a": "b"}'},
                header: {name: 'fieldname', invisible: false},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListJson, {props: {
                row: {fieldname: '{"a": "b"}', invisible: true},
                header: {name: 'fieldname', invisible: 'fields.invisible'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListJson, {props: {
                row: {fieldname: '{"a": "b"}', invisible: false},
                header: {name: 'fieldname', invisible: 'fields.invisible'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('Json Thumbnail component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render with data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {fieldname: '{"a": "b"}'},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {fieldname: '{"a": "b"}'},
                name: 'fieldname',
                invisible: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {fieldname: '{"a": "b"}'},
                name: 'fieldname',
                invisible: false,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {fieldname: '{"a": "b"}', invisible: true},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {fieldname: '{"a": "b"}', invisible: false},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render label', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {fieldname: '{"a": "b"}'},
                name: 'fieldname',
                label: 'The label',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip up', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {fieldname: '{"a": "b"}'},
                name: 'fieldname',
                tooltip: 'The tooltip',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip left', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailJson, {props: {
                data: {fieldname: '{"a": "b"}'},
                name: 'fieldname',
                tooltip: 'The left tooltip',
                tooltip_position: 'is-left',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('Json Form component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render with data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {}},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}},
                name: 'fieldname',
                invisible: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}},
                name: 'fieldname',
                invisible: false,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}', invisible: true}},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}', invisible: false}},
                name: 'fieldname',
                invisible: 'fields.invisible',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render label', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}},
                name: 'fieldname',
                label: 'The label',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip up', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}},
                name: 'fieldname',
                tooltip: 'The tooltip',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip left', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}},
                name: 'fieldname',
                tooltip: 'The left tooltip',
                tooltip_position: 'is-left',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly by config', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}, mode: 'readonly'},
                name: 'fieldname',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly param', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}},
                name: 'fieldname',
                readonly: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render mode readonly conditional param', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}', readonly: 1}},
                name: 'fieldname',
                readonly: 'fields.readonly',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render required', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}'}},
                name: 'fieldname',
                required: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render required conditionnal', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {fieldname: '{"a": "b"}', required: true}},
                name: 'fieldname',
                required: 'fields.required',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render required without value', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {}},
                name: 'fieldname',
                required: true,
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render placeholder', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldFormJson, {props: {
                config: {data: {}},
                name: 'fieldname',
                placeholder: 'The placeholder'
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});
