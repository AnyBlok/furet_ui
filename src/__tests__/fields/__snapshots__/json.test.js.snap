// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Json Form component Render invisible condition is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="{
  &quot;a&quot;: &quot;b&quot;
}" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <!----></div></span>
`;

exports[`Json Form component Render invisible condition is true 1`] = `<div data-server-rendered="true"></div>`;

exports[`Json Form component Render invisible is True 1`] = `<div data-server-rendered="true"></div>`;

exports[`Json Form component Render invisible is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="{
  &quot;a&quot;: &quot;b&quot;
}" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <!----></div></span>
`;

exports[`Json Form component Render label 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><label class="label">The label</label> <textarea rows="1" value="{
  &quot;a&quot;: &quot;b&quot;
}" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <!----></div></span>
`;

exports[`Json Form component Render mode readonly by config 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Form component Render mode readonly conditional param 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Form component Render mode readonly param 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Form component Render placeholder 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea placeholder="The placeholder" rows="1" value="" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <p class="help is-danger">Invalid JSON format</p></div></span>`;

exports[`Json Form component Render required 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="{
  &quot;a&quot;: &quot;b&quot;
}" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <!----></div></span>
`;

exports[`Json Form component Render required conditionnal 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="{
  &quot;a&quot;: &quot;b&quot;
}" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <!----></div></span>
`;

exports[`Json Form component Render required without value 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <p class="help is-danger">Invalid JSON format</p></div></span>`;

exports[`Json Form component Render tooltip left 1`] = `
<span data-label="The left tooltip" data-server-rendered="true" class="tooltip is-primary is-left is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="{
  &quot;a&quot;: &quot;b&quot;
}" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <!----></div></span>
`;

exports[`Json Form component Render tooltip up 1`] = `
<span data-label="The tooltip" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="{
  &quot;a&quot;: &quot;b&quot;
}" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <!----></div></span>
`;

exports[`Json Form component Render with data 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="{
  &quot;a&quot;: &quot;b&quot;
}" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <!----></div></span>
`;

exports[`Json Form component Render without data 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium" style="width:100%;"><div class="field" style="width:inherit;"><!----> <textarea rows="1" value="" class="textarea" style="width:100%;height:100%;resize:none;"></textarea> <p class="help is-danger">Invalid JSON format</p></div></span>`;

exports[`Json Thumbnail component Render invisible condition is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Thumbnail component Render invisible condition is true 1`] = `<div data-server-rendered="true"></div>`;

exports[`Json Thumbnail component Render invisible is True 1`] = `<div data-server-rendered="true"></div>`;

exports[`Json Thumbnail component Render invisible is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Thumbnail component Render label 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><label class="label">The label</label> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Thumbnail component Render tooltip left 1`] = `
<span data-label="The left tooltip" data-server-rendered="true" class="tooltip is-primary is-left is-medium"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Thumbnail component Render tooltip up 1`] = `
<span data-label="The tooltip" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Thumbnail component Render with data 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre> <!----></div></span>
`;

exports[`Json Thumbnail component Render without data 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <pre style="width:100%;padding:2;background-color:white;"></pre> <!----></div></span>`;

exports[`Json list component Render invisible condition is false 1`] = `
<pre data-server-rendered="true" style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre>
`;

exports[`Json list component Render invisible condition is true 1`] = `<span data-server-rendered="true"></span>`;

exports[`Json list component Render invisible is True 1`] = `<span data-server-rendered="true"></span>`;

exports[`Json list component Render invisible is false 1`] = `
<pre data-server-rendered="true" style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre>
`;

exports[`Json list component Render with data 1`] = `
<pre data-server-rendered="true" style="width:100%;padding:2;background-color:white;">{
  &quot;a&quot;: &quot;b&quot;
}</pre>
`;

exports[`Json list component Render without data 1`] = `<pre data-server-rendered="true" style="width:100%;padding:2;background-color:white;"></pre>`;
